src/responder/pam/pamsrv_json.c

```diff
@@ -186,9 +224,227 @@ obtain_prompts(struct confdb_ctx *cdb, TALLOC_CTX *mem_ctx,
         }
     }
 
+    if (sc_init_prompt == NULL) {
```

Comment:
Do you need to check for an empty string here also? In `pc_list_add_smartcard()` you have 

```
    pc->data.smartcard.prompt_init = strdup(prompt_init != NULL ? prompt_init
                                            : "");
    if (pc->data.smartcard.prompt_init == NULL) {
        ret = ENOMEM;
        goto done;
    }
```

Reply:
I think that this part of the code is more complex than necessary now that we only allow hardcoded strings. That's probably what is nailing you. Do you think it'd be wiser to assign the value with `talloc_strdup()` without checking whether the prompts are NULL? In all cases they are NULL so that wouldn't hurt.